radix-engine/src/kernel/kernel.rs

```diff
@@ -215,6 +215,27 @@ struct KernelHandler<
     callback: &'a mut M,
     prev_frame: Option<&'a CallFrame<M::CallFrameData, M::LockData>>,
     on_store_access: F,
+    read_from_heap: bool,
```

Comment:
Can we just combine this with `on_store_access` so that `StoreAccess` is repaced with a bigger enum. For example,
```
enum CallFrameEvent {
    StoreAccess(StoreAccess),
    HeapAccess,
}
```

Reply:
Yes, actually such a solution I had before, but after some heavy changes on develop around that enum I switched to boolean, but I will change that.

Reply:
I've extended `StoreAccess` enum with dedicated heap read variant, because it is more straightforward in my opinion than modification of `KernelHandler` (that will also require to add another event handler for heap store access).